internal/util/k8s/client.go

```diff
@@ -56,3 +56,11 @@ func NewK8sClient() (*kubernetes.Clientset, error) {
 
 	return client, nil
 }
+
+// RunsOnKubernetes checks if the application is running within a Kubernetes cluster
+// by inspecting the presence of the KUBERNETES_SERVICE_HOST environment variable.
+func RunsOnKubernetes() bool {
```

Comment:
yay, auto-detection!

Can you move this commit before the one that calls this function? It should be possible to at least build the ceph-csi executable after each commit (otherwise `git bisect` breaks).

Reply:
Done